drivers/spi/spi_ll_stm32.c

```diff
@@ -744,6 +799,13 @@ static int transceive_dma(const struct device *dev,
 		return -ENOTSUP;
 	}
 
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+	if ((tx_bufs != NULL && !spi_buf_set_in_nocache(tx_bufs)) ||
```

Comment:
I think the condition `tx_bufs != NULL`  should be checked by `spi_buf_set_in_nocache` to do a bit of parameters sanitation.

Reply:
That makes things more confusing in my eyes. It would require `spi_buf_set_in_nocache` tu return an error code instead of a `bool` to identify when the buffers are, or are not, in nocache, and when the input parameter is an error. That's not the intention with that function, I want to to return a `bool` to be coherent and simple.

Also, it obscures the semantic of that `tx_bufs` and `rx_bufs` should only be checked if the SPI transaction involves read and/or write respectively. Checking parameters inside that function tells that `tx_bufs` need to be checked even when the SPI operation is read only, or that `rx_bufs` need to be checked when the operation is write only. I want to make it clear that they only need to be checked if the operations requires so.

Finally, a `NULL` is essentially address = 0x0. That technically is an address as valid as any other one to be checked to verify if it is nocache or not. Providing a special treatment to it confuses what the function does.

Reply:
fair enough